spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1466,245 +1470,216 @@ def handle_max_vel(self, request: SetVelocity.Request, response: SetVelocity.Res
+            or feedback.status.value == feedback.status.STATUS_COMMAND_TIMED_OUT
+            or feedback.status.value == feedback.status.STATUS_ROBOT_FROZEN
+            or feedback.status.value == feedback.status.STATUS_INCOMPATIBLE_HARDWARE
+        ):
+            return GoalResponse.FAILED
+
+        fb = feedback.feedback
+        choice = fb.feedback_choice
+
+        if choice == fb.FEEDBACK_SE2_TRAJECTORY_FEEDBACK_SET:
+            if fb.se2_trajectory_feedback.status.value != fb.se2_trajectory_feedback.status.STATUS_AT_GOAL:
+                return GoalResponse.IN_PROGRESS
+        elif choice == fb.FEEDBACK_SE2_VELOCITY_FEEDBACK_SET:
+            self.get_logger().warn("WARNING: Planar velocity commands provide no feedback")
+        elif choice == fb.FEEDBACK_SIT_FEEDBACK_SET:
+            if fb.sit_feedback.status.value != fb.sit_feedback.status.STATUS_IS_SITTING:
                 return GoalResponse.IN_PROGRESS
-            elif (
-                full_body_feedback.feedback.feedback_choice
-                == full_body_feedback.feedback.FEEDBACK_PAYLOAD_ESTIMATION_FEEDBACK_SET
+        elif choice == fb.FEEDBACK_STAND_FEEDBACK_SET:
+            if fb.stand_feedback.status.value != fb.stand_feedback.status.STATUS_IS_STANDING:
+                return GoalResponse.IN_PROGRESS
+        elif choice == fb.FEEDBACK_STANCE_FEEDBACK_SET:
+            if fb.stance_feedback.status.value == fb.stance_feedback.status.STATUS_TOO_FAR_AWAY:
+                return GoalResponse.FAILED
+            if fb.stance_feedback.status.value != fb.stance_feedback.status.STATUS_STANCED:
+                return GoalResponse.IN_PROGRESS
+        elif choice == fb.FEEDBACK_STOP_FEEDBACK_SET:
+            self.get_logger().warn("WARNING: Stop command provides no feedback")
+        elif choice == fb.FEEDBACK_FOLLOW_ARM_FEEDBACK_SET:
+            self.get_logger().warn("WARNING: FollowArmCommand provides no feedback")
+        elif choice == fb.FEEDBACK_NOT_SET:
+            # sync_feedback.mobility_command_feedback_is_set, feedback_choice is actually not set.
+            # This may happen when a command finishes, which means we may return SUCCESS below.
+            self.get_logger().info("mobility command feedback indicates goal has reached")
+        else:
+            self.get_logger().error("ERROR: unknown mobility command type")
+            return GoalResponse.IN_PROGRESS
+        return None
+
+    def _process_synchronized_gripper_command_feedback(
+        self, feedback: GripperCommandFeedback
+    ) -> Optional[GoalResponse]:
+        if (
+            feedback.status.value == feedback.status.STATUS_COMMAND_OVERRIDDEN
+            or feedback.status.value == feedback.status.STATUS_COMMAND_TIMED_OUT
+            or feedback.status.value == feedback.status.STATUS_ROBOT_FROZEN
+            or feedback.status.value == feedback.status.STATUS_INCOMPATIBLE_HARDWARE
+        ):
+            return GoalResponse.FAILED
```

Comment:
Should this if statement be moved to a function somewhere?

Reply:
Good call, I'll pull this out to make it its own function.